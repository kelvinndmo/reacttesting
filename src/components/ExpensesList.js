import React from "react";
import { connect } from "react-redux";
import ExpenseListItem from "./ExpenseListItem";
import SelectedExpense from "../selectors/expenses";

export const ExpenseList = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>kelvin onkundi ndemo</p>
    ) : (
      props.expenses.map(expense => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
    {}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: state.expenses,
  };
};
export default connect(mapStateToProps)(ExpenseList);
