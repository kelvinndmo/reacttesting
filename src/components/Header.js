import React from 'react'
import { NavLink, Link } from 'react-router-dom'

const Header = () => (
  <header>
    <h1>Expensify</h1>
    <NavLink to='/' activeClassName='is-active' exact>
      Home
    </NavLink>
    <NavLink to='/create' activeClassName='is-active'>
      Create Expense
    </NavLink>
  </header>
)

export default Header
