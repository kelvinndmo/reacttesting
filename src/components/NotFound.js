import React from "react";
import { Link } from "react-router-dom";

const NotFound = () => (
  <div>
    <h3>
      404! <Link to="/">- The page ure looking for is unavaibale</Link>
    </h3>
  </div>
);

export default NotFound;
