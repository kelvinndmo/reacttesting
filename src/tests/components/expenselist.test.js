import React from "react";
import { shallow } from "enzyme";
import { ExpenseList } from "../../components/ExpensesList";
import { ExpenseListItem } from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should render ExpenseList with expenses", () => {
  const wrapper = shallow(<ExpenseList expenses={expenses} />);
  expect(wrapper).toMatchSnapshot();
});

test("should get a message when there are no items", () => {
  const wrapper = shallow(<ExpenseList expenses={[]} />);
  expect(wrapper).toMatchSnapshot();
});
