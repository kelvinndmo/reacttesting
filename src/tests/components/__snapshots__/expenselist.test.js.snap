// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get a message when there are no items 1`] = `
<div>
  <p>
    kelvin onkundi ndemo
  </p>
</div>
`;

exports[`should render ExpenseList with expenses 1`] = `
<div>
  <Connect(ExpenseListItem)
    amount={50}
    createdAt={0}
    description="kelvin"
    id="1 "
    key="1 "
    note="back 95"
  />
  <Connect(ExpenseListItem)
    amount={50000}
    createdAt={-345600000}
    description="kelvin onkundi"
    id="2"
    key="2"
    note="back 95"
  />
  <Connect(ExpenseListItem)
    amount={5000}
    createdAt={345600000}
    description="kelvin ndemo"
    id="3"
    key="3"
    note="back 95"
  />
</div>
`;
